src/Npgsql/NpgsqlConnection.cs

```diff
@@ -134,15 +134,17 @@ public INpgsqlTypeMapper TypeMapper
         /// Initializes a new instance of the
         /// <see cref="NpgsqlConnection">NpgsqlConnection</see> class.
         /// </summary>
-        public NpgsqlConnection()
-            => GC.SuppressFinalize(this);
+        public NpgsqlConnection() : this("") { }
```

Comment:
The reason I made this change, is that in normal applications it is sometimes common to first instantiate the NpgsqlConnection, and then set the ConnectionString property (this is mandatory when using NpgsqlFactory). This meant that the ConnectionString setter was called twice - once from the empty constructor and once when setting it, and this was hurting efficiency (the pool was looked up twice internally etc.)

What exactly is the issue with the VSIX, can you isolate the behavior it requires? I'm sure it's possible to fix this without going back to two lookups.

By the way, I just noticed that when the constructor with the connection string is called, `GC.SuppressFinalize()` wasn't being called - this should also be fixed.